cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3
+
+# Contest Management System - http://cms-dev.github.io/
+# Copyright © 2018 Pēteris Pakalns <peterispakalns@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import io
+import os
+import re
+import logging
+import yaml
+import zipfile
+import datetime
+import subprocess
+
+from cms import config
+from cms.db import Contest, Dataset, Task, Statement, Testcase, Manager
+from .base_loader import ContestLoader, TaskLoader
+from cms import TOKEN_MODE_DISABLED, TOKEN_MODE_FINITE, TOKEN_MODE_INFINITE
+from cmscommon.constants import \
+    SCORE_MODE_MAX, SCORE_MODE_MAX_SUBTASK, SCORE_MODE_MAX_TOKENED_LAST
+from .italy_yaml import load_yaml_from_path, make_timedelta
+from datetime import timedelta
+
+logger = logging.getLogger(__name__)
+
+
+def set_if_present(src_dict, trg_dict, key, conv=None, default=None):
+    if key in src_dict:
+        if conv:
+            trg_dict[key] = conv(src_dict[key])
+        else:
+            trg_dict[key] = src_dict[key]
+    elif default is not None:
+        trg_dict[key] = default
+
+
+class LioTaskLoader(TaskLoader):
+
+    short_name = "Lio-task"
+    description = "Latvian Informatics Olympiad task loader"
+
+    def __init__(self, path, file_cacher):
+        super().__init__(path, file_cacher)
+        self.task_dir = os.path.dirname(self.path)
+        self.conf = load_yaml_from_path(self.path)
+
+
+    @staticmethod
+    def detect(path):
+        # TODO: Support auto detection
+        return False
+
+
+    def parse_point_file(self, point_path):
```

Comment:
I agree, but we probably need to introduce this format for importing tasks first and then we can ask other members to switch their group point definition way to the yaml file and it would be nice if they would write the yaml file too. (We probably still will need to fix it) At least now it is more simple for them.
But if we standardize the layout of files there wouldn't be a lot of fields to specify for them.